components/server/ee/src/prebuilds/prebuild-manager.ts

```diff
@@ -228,4 +238,33 @@ export class PrebuildManager {
         }
         return hostContext.contextParser;
     }
-}
\ No newline at end of file
+
+    private async shouldRateLimitPrebuild(span: opentracing.Span, cloneURL: string): Promise<boolean> {
+        const minuteAgo = secondsBefore(new Date().toISOString(), 60)
+        const unabortedCount = await this.workspaceDB.trace({span}).countUnabortedPrebuildsSince(cloneURL, new Date(minuteAgo))
+        const limit = this.getPrebuildRateLimitForCloneURL(cloneURL)
+
+        if (unabortedCount >= limit) {
+            log.debug("Prebuild exceeds rate limit", { limit, unabortedPrebuildsCount: unabortedCount, cloneURL })
+            return true;
+        }
+        return false;
+    }
+
+    private getPrebuildRateLimitForCloneURL(cloneURL: string): number {
+        // First we use any explicit overrides for a given cloneURL
+        let limit = Number(this.config.prebuildLimiter[cloneURL]);
```

Comment:
Alright, scrape the type, just found [it](https://github.com/gitpod-io/gitpod/pull/8568/files#diff-b53cd6a3e7c6f723a548f8c5560dc26b34591c201f4605cdf7c6ca1ced8e651bR158). :+1:

Reply:
So the type is pushed in there, but the type system doesn't guarantee that the value that's in the config is in fact a number. For that reason, the extra guards to ensure that we only operate with numbers.

Reply:
True. What I wanted to get at is, that we typically don't do this kind of low-level verification. If we were to introduce type-verification it would be a schema for the config.

If you adjust the type as suggested in the other comment, at least the compiler is happy - and that'd be fine with me as well.

Reply:
Sure, happy to make that change.